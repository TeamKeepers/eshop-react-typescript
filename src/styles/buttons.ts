import styled from "@emotion/styled";

export const CatchyBtn = styled.button`
    width: 150px;
    height: 50px;
    font-size: 0.8em;
    background: ${props => props.disabled ? 'lightgrey' : props.theme.colors.catchy};
    color: white;
    border: none;
    border-radius: ${props => props.theme.radius};
    cursor: ${props => props.disabled ? '' : 'pointer'};
    outline: none;
    :hover {
       background: ${props => props.disabled ? '' : props.theme.colors.main};
    }
    :active {
      transform: translateY(4px);
    }
`;

export const PrimaryBtn = styled.button`
    width: auto;
    height: 35px;
    margin-top: 20px;
    font-size: 0.6em;
    background: ${props => props.disabled ? 'lightgrey' : props.theme.colors.success};
    color: white;
    border: none;
    border-radius: ${props => props.theme.radius};
    cursor: ${props => props.disabled ? '' : 'pointer'};
    outline: none;
    :hover {
       background: ${props => props.disabled ? '' : props.theme.colors.main};
    }
    :active {
      transform: translateY(4px);
    }
`;

export const LoadBtn = styled.button`
    width:90px;
    height: 40px;
    margin-top: 20px;
    font-size: 0.5em;
    background: ${props => props.disabled ? 'lightgrey' : props.theme.colors.fade};
    color: #fff;
    border: none;
    border-radius: ${props => props.theme.radius};
    cursor: ${props => props.disabled ? 'not-allowed' : 'pointer'};
    outline: none;
    :hover {
       background: ${props => props.disabled ? '' : props.theme.colors.main};
    }
    :active {
      transform: translateY(4px);
    }
`;

export const GoBackBtn = styled.button`
    color: ${props => props.theme.colors.fade};
    font-size: 0.5em;
    background-color: #fff;
    border: none;
    width: auto;
    max-height: 20px;
    margin: 30px 0;
    display: inline-flex;
    align-items: center; 
    cursor: ${props => props.disabled ? '' : 'pointer'};
`;