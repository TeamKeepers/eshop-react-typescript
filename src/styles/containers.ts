import styled from "@emotion/styled";

export const PageContainer = styled.section`
    width: 100%;
    height: 100vh;
    display: flex;
    flex-direction: column;
    align-items: flex-start;
`;

export const MainContainer = styled.main`
    width: 100%;
    min-height: 86vh;
    display: flex;
    justify-content: center;
    overflow: scroll;
    margin-bottom: 50px;
`;

export const BasicContainer = styled.section`
    width: 80%;
    display: flex;
    flex-wrap: wrap;
    justify-content: space-evenly;
`;

export const LoadMoreContainer = styled.section`
    width: 100%;
    display: flex;
    justify-content: center;
`;

export const SingleProductContainer = styled.section`
    width: 80%;
    display: flex;
    flex-direction: column;
    align-items: flex-start;
`;

export const CardContainer = styled.div`
    width: 30%;
    box-shadow: ${props => props.theme.boxShadow};
    border: 0.5px ${props => props.theme.colors.fade} solid;
    margin-top: 25px;
    padding-bottom: 15px;
    max-height: 45vh;

    @media screen and (max-width: 1000px) {
        width: 45%;
    }

    @media screen and (max-width: 700px) {
        width: 100%;
        min-height: 52vh;
    }

`;


export const FormContainer = styled.form`
    width: 100%;
    display: flex;
    align-items: center;
    justify-content: space-evenly;
    font-size: 0.6em;

    .input {
        display: flex;
        flex-direction: column;
    }
`