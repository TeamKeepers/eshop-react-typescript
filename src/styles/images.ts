import styled from "@emotion/styled";

export const HeaderIcons = styled.img`
    width: 50px;
    margin: 0 5%;
    height: 100%;
    `;

export const TinyIcons = styled.img`
    width: 23px;
    margin: 0 8px;
    `;

export const BasicImg = styled.img`
    width: 220px;

    @media screen and (max-width: 700px) {
        padding-top: 50px;
    }
`;

export const ProductImg = styled.img`
    width: 100%;
    height: 20vh;
    object-fit: cover;
    object-position: center center;
`;

export const LoadingImg = styled.img`
    width: 50%;
    height: 50vh;

    @media screen and (max-width: 700px) {
        width: 100%;
    }
`;

export const SingleProductImg = styled.img`
    width: 50%;
    object-fit: cover;
    object-position: center center;

    @media screen and (max-width: 1000px) {
        width: 100%;
    }
`;

export const CartProductImg = styled.img`
    width: 40%;
    object-fit: cover;
    object-position: center center;
`;
