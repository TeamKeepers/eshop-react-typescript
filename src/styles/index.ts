/**
 * DEFINE A THEM IS A PAIN IN THE ASS WITH EMOTION
 * SOMETIMES WORKING, SOMETIMES NOT
 */
import '@emotion/react'

declare module '@emotion/react' {
    export interface Theme {
        colors: {
            main: string,
            success: string,
            catchy: string,
            fade: string,
        },
        boxShadow: string,
        radius: string
    }
}

// export type ThemeType = {
//     colors: {
//         main: string,
//         success: string,
//         catchy: string,
//         fade: string,
//     },
//     boxShadow: string,
//     radius: string
// }

export const theme = {
    colors: {
        main: "#000",
        success: "#62c16c",
        catchy: "#d80027",
        fade: "rgba(0, 0, 0, .2)",
    },
    boxShadow: "5px 5px 0.2px 0.5px rgba(0, 0, 0, .2)",
    radius: "5px"
};