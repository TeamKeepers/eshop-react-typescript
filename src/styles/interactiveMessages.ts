import styled from "@emotion/styled";

export const ErrorMessage = styled.div`
    width: 100%;
    margin-top: 20px;
    padding: 20px;
    font-size: 0.8em;
    border-radius: ${props => props.theme.radius};
    color: #fff;
    display: flex;
    justify-content: center;
    align-items: center;
    background-color: ${props => props.theme.colors.catchy};
`;