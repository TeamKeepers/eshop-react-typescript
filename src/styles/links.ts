import styled from "@emotion/styled";
import { Link } from "react-router-dom";

export const BasicLink = styled(Link)`
    text-decoration: none;
`;

export const CardLink = styled(Link)`
    text-decoration: none;
    color: ${props => props.theme.colors.main};
`;

export const CartIconLink = styled(Link)`
    text-decoration: none;
    color: ${props => props.theme.colors.main};
    position: relative;

    #pin {
        background-color: ${props => props.theme.colors.catchy};
        display: flex;
        justify-content: center;
        align-items: center;
        height: 20px;
        width: 20px;
        font-size: 0.5em;
        color: #fff;
        border-radius: 100%; 
        position: absolute;
        right: -10px;
        top: 8px;
    }    
`;

