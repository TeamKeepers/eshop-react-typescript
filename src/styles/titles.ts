import styled from "@emotion/styled";

export const Title = styled.h1`
    color: #fff;
    font-size: 1.8em;

    span {
        background-color: #000;
    }

    @media screen and (max-width: 700px) { 
        font-size: 1.3em;
    }

`;

export const SecondTitle = styled.h2`
    color: #fff;
    font-size: 1.4em;

    span {
        background-color: #000;
    }

    @media screen and (max-width: 700px) { 
        font-size: 1em;
        text-align: center;
    }

`;

export const CardTitle = styled.h3`
    font-size: 0.5em;
    margin-bottom: 10px;
`;

export const CategoryTitle = styled.h5`
    color: #fff;
    font-size: 0.4em;
    margin-bottom: 10px;

    span {
        background-color: #000;
    }

`;

export const CategoryTitleSingleProduct = styled.h5`
    color: #fff;
    font-size: 0.8em;
    margin-bottom: 10px;

    span {
        background-color: #000;
    }

`;

export const CardTitleSingleProduct = styled.h3`
    font-size: 0.8em;
    margin-bottom: 10px;
`;

export const PriceCartTitle = styled.h3`
    color: #fff;
    font-size: 0.9em;
    margin-bottom: 10px;
    
    span {
        background-color: #000;
    }

`; 