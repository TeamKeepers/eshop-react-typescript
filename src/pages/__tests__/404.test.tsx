import React from "react";
import { render, cleanup } from "@testing-library/react";
import Page404 from "../404";
import { ThemeProvider } from "@emotion/react";
import { theme } from "../../styles";

afterEach(cleanup);

it("<Page404 /> snapshot", () => {        
    const { container } = render(
        <ThemeProvider theme={theme}>
            <Page404 />
        </ThemeProvider>
    );
    expect(container.firstChild).toMatchSnapshot();
})