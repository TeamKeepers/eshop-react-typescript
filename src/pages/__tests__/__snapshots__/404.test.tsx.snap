// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Page404 /> snapshot 1`] = `
<section
  class="css-h3xh94"
>
  <img
    alt="404"
    class="css-wbx0un"
    src="404.gif"
  />
  <h2>
    Woopsy
  </h2>
  <p>
    Lost, aren't you ?
  </p>
  <button
    class="css-lyx6qd"
  >
    Let's go back
  </button>
</section>
`;
